src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1511,6 +1529,56 @@ public bool UsePerfCounters
 
         #region Misc
 
+        internal void InitServiceFile()
```

Comment:
From a factoring perspective, it may be better to take out this implementation into a separate PgConnectionServiceFile (like we have PgPassFile). Depending on the answers to the questions above, it may also be better to use it from NpgsqlConnector.Open rather than in GetPoolAndSettings.